VisaCheckout/src/main/java/com/braintreepayments/api/visacheckout/VisaCheckoutClient.kt

```diff
@@ -0,0 +1,153 @@
+package com.braintreepayments.api.visacheckout
+
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import com.braintreepayments.api.core.ApiClient
+import com.braintreepayments.api.core.BraintreeClient
+import com.braintreepayments.api.core.Configuration
+import com.braintreepayments.api.core.ConfigurationException
+import com.braintreepayments.api.core.TokenizeCallback
+import com.braintreepayments.api.visacheckout.VisaCheckoutNonce.Companion.fromJSON
+import com.visa.checkout.Environment
+import com.visa.checkout.Profile
+import com.visa.checkout.Profile.ProfileBuilder
+import com.visa.checkout.VisaPaymentSummary
+import org.json.JSONException
+import org.json.JSONObject
+
+/**
+ * Used to create and tokenize Visa Checkout. For more information see the [documentation](https://developer.paypal.com/braintree/docs/guides/secure-remote-commerce/overview)
+ */
+class VisaCheckoutClient internal constructor(
+    private val braintreeClient: BraintreeClient,
+    private val apiClient: ApiClient = ApiClient(braintreeClient)
+) {
+    /**
+     * Initializes a new [VisaCheckoutClient] instance
+     *
+     * @param context an Android Context
+     * @param authorization a Tokenization Key or Client Token used to authenticate
+     */
+    constructor(context: Context, authorization: String) : this(
+        BraintreeClient(context, authorization)
+    )
+
+    /**
+     * Creates a [Profile.ProfileBuilder] with the merchant API key, environment, and other
+     * properties to be used with Visa Checkout.
+     *
+     * In addition to setting the `merchantApiKey` and `environment` the other properties that
+     * Braintree will fill in on the ProfileBuilder are:
+     *
+     * [Profile.ProfileBuilder.setCardBrands] A list of Card brands that your merchant account can
+     * transact.
+     *
+     * [Profile.ProfileBuilder.setDataLevel] - Required to be [Profile.DataLevel.FULL] for Braintree to
+     * access card details
+     *
+     * [Profile.ProfileBuilder.setExternalClientId] -  Allows the encrypted payload to be processable
+     * by Braintree.
+     *
+     * @param callback [VisaCheckoutCreateProfileBuilderCallback]
+     */
+    fun createProfileBuilder(
+        callback: VisaCheckoutCreateProfileBuilderCallback
+    ) {
+        braintreeClient.getConfiguration { configuration: Configuration?, exception: Exception? ->
+            if (configuration != null) {
```

Comment:
Added a null check here. Prior to converting to Kotlin, there would have been a crash trying to access properties on `configuration`.